fastecdsa/ecdsa.py

```diff
@@ -28,14 +28,19 @@ def sign(msg: MsgTypes, d: int, curve: Curve = P256, hashfunc=sha256, prehashed:
         |  hashfunc (_hashlib.HASH): The hash function used to compress the message.
         |  prehashed (bool): The message being passed has already been hashed by :code:`hashfunc`.
     """
-    # generate a deterministic nonce per RFC6979
-    rfc6979 = RFC6979(msg, d, curve.q, hashfunc, prehashed=prehashed)
-    k = rfc6979.gen_nonce()
-
     if prehashed:
-        hex_digest = hexlify(msg).decode()
+        try:
+            hex_digest = msg.hexdigest()
+            msg = msg.digest()
+        except:
+            hex_digest = hexlify(msg).decode()
```

Comment:
This breaks the interface to the function by allowing either a byte string or a hash object to be passed. I'd prefer we keep the interface simple and scope it to just byte strings. That means you'll have to pass the result of `digest` in here.

(To be quite honest, I strongly dislike `prehashed` to begin with, it's a super annoying feature, it doesn't provide any functionality other than compatibility with other libs that allow pre-hashing (why???), and it causes all sorts of bugs because there is no spec for prehashed ECDSA so everyone implements it differently).

Reply:
Of course, I just add extra syntactic sugar constantly :)

It's not about "compatibility", i didn’t even think about other libraries. It just seemed strange and inconvenient to me that you can use a hash to sign, but you need to have the original message when verifying. If the message is too long - it will be inconvenient to store it.